test/foundry/src/concrete/vault/OffchainAssetReceiptVault.handler.t.sol

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: CAL
+pragma solidity =0.8.25;
+
+import {
+    OffchainAssetReceiptVault,
+    OffchainAssetReceiptVaultConfig
+} from "../../../../../contracts/concrete/vault/OffchainAssetReceiptVault.sol";
+import {OffchainAssetReceiptVaultTest, Vm} from "test/foundry/abstract/OffchainAssetReceiptVaultTest.sol";
+import {LibOffchainAssetVaultCreator} from "test/foundry/lib/LibOffchainAssetVaultCreator.sol";
+import {Receipt as ReceiptContract} from "../../../../../contracts/concrete/receipt/Receipt.sol";
+
+contract OffchainAssetReceiptVaultHandlerTest is OffchainAssetReceiptVaultTest {
+    event SetERC20Tier(address sender, address tier, uint256 minimumTier, uint256[] context, bytes data);
+    event SetERC1155Tier(address sender, address tier, uint256 minimumTier, uint256[] context, bytes data);
+    event OffchainAssetReceiptVaultInitialized(address sender, OffchainAssetReceiptVaultConfig config);
+    event DepositWithReceipt(
+        address sender, address owner, uint256 assets, uint256 shares, uint256 id, bytes receiptInformation
+    );
+
+    /// Get Receipt from event
+    function getReceipt() internal returns (ReceiptContract) {
```

Comment:
should we put this on the parent `OffchainAssetReceiptVaultTest` contract and use it everywhere that we need a receipt in any receipt test?